### PR TITLE
dedir: Add new pallet for `Decentralised Directories` with `blob`

### PR DESCRIPTION
**Pull request by vatsa287**
Updates: #503

Abandoned. Do not merge.

### CODE REVIEW

File: pallets/dedir/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/// The `Permissions` enum defines the levels of access control available for an account within a
+/// registry.
+///
+/// - `DELEGATE`: Grants permission to manage registry entries.
+/// - `ADMIN`: Extends `DELEGATE` permissions, allowing the management of delegates in addition to
+///   managing registry entries.
+/// - `OWNER`: The creator or owner of the registry. This permission level encompasses the full
+///   range of management capabilities, including the permissions of both `DELEGATE` and `ADMIN`.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo)]
+pub enum Permissions {
```
Comment (smohan-dw) [inline pallets/dedir/src/types.rs]: Permission as Enums will not help to create composite permissions. A delegate can have multiple roles associated with a registry. Please refer the chain-space permissioning system.

File: pallets/dedir/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+//! DeDir pallet types.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+
+use sp_runtime::RuntimeDebug;
+use sp_std::prelude::*;
+
+#[derive(Encode, Decode, MaxEncodedLen, Clone, RuntimeDebug, PartialEq, Eq, TypeInfo)]
+pub enum RegistrySupportedStateOf {
```
Comment (smohan-dw) [inline pallets/dedir/src/types.rs:32]: Do we need a DRAFT state at the chain level? Without that this can be a bool filed representing the status

Comment (amarts) [reply pallets/dedir/src/types.rs:32]: This thought process came when we asked the question if chain alone is used as 'store' also, how to differentiate between something which is not ready to be 'lookedup' (or queried), and the one record which is ready to do the lookup.

Having these states covers most of the usecases is what we thought.

File: pallets/dedir/src/types.rs (hunk 1 above)
Comment (vatsa287) [reply pallets/dedir/src/types.rs]: Ack,
Have updated the flow to use bitwise and handle composite permissions.

**Comment by amarts** [comment]
done with #504

**Pull request closed without merging**